crates/derive/src/pipeline/core.rs

```diff
@@ -56,8 +56,13 @@ where
     S: NextAttributes + ResettableStage + OriginProvider + OriginAdvancer + Debug + Send + Sync,
     P: L2ChainProvider + Send + Sync + Debug,
 {
+    /// Peeks at the next prepared [L2AttributesWithParent] from the pipeline.
+    fn peek(&self) -> Option<&L2AttributesWithParent> {
+        self.prepared.front()
+    }
+
     /// Returns the next prepared [L2AttributesWithParent] from the pipeline.
-    fn next_attributes(&mut self) -> Option<L2AttributesWithParent> {
+    fn next(&mut self) -> Option<L2AttributesWithParent> {
```

Comment:
Not necessary for this PR, but what do you think about just making this an [`Iterator`](https://doc.rust-lang.org/stable/core/iter/trait.Iterator.html) + [`Peekable`](https://doc.rust-lang.org/stable/core/iter/struct.Peekable.html) later? Follows the same API as we have here w/ existing traits from `core`.